lib/crawlexir/search/result_page.ex

```diff
@@ -0,0 +1,82 @@
+defmodule Crawlexir.Search.ResultPage do
+  defstruct advertising_content: %{}, organic_content: %{}, page_content: %{}, raw_html: nil
+
+  alias Crawlexir.Search.ResultPage
+
+  @advertiser_selector "li.ads-ad"
+  @advertiser_link_selector ".ad_cclk a.V0MxL"
+  @organic_link_selector "#search .g .r > a"
+
+  @doc """
+  Parse a Google search page and retrieve advertising, organic and general page content.
+
+  ## Example
+
+      iex> new("<html><head>< ...")
+      {:ok,
+       %ResultPage{
+        advertising_content: %{
+          count: 1,
+          url_list: ["https://some-links.com/"]
+        },
+        organic_content: %{
+          count: 1,
+          url_list: ["https://some-links.com/"]
+        },
+        page_content: %{link_count: 100},
+        raw_html:  "<!doctype html><html> <> ..."
+      }
+  """
+  def new(body) do
```

Comment:
I am thinking about maybe we can name it `parse`. When I saw new sometimes I'm thinking of a new object.

Reply:
I am still unsure about that as well. I saw some lib using `new` but yes I also feel that `parse` would feel more functional programming oriented. Changed in f88d0052048318fa7db2dd3a4a16fd9af1c4190a